### PR TITLE
[unipool-same-token] Add unipool-same-token strategy

### PR DESCRIPTION
**Pull request by dapplion**
Generic strategy for UniPool contract where the staked token equals the earned token

### CODE REVIEW

File: src/strategies/index.ts [hunk 1]
```diff
@@ -227,5 +228,7 @@ export default {
   'planet-finance': planetFinance,
   'impossible-finance': impossibleFinance,
   badgeth,
-  'tranche-staking-slice': trancheStakingSLICE
+  'impossible-finance': impossibleFinance,
```
Comment (ChaituVR) [inline src/strategies/index.ts:231]: plz remove this line

**Review by ChaituVR — CHANGES_REQUESTED** [review]

File: src/strategies/index.ts (hunk 1 above)
Comment (dapplion) [reply src/strategies/index.ts]: Done

File: src/strategies/unipool-same-token/index.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  {
+    constant: true,
+    inputs: [{ internalType: 'address', name: 'account', type: 'address' }],
+    name: 'earned',
+    outputs: [{ internalType: 'uint256', name: '', type: 'uint256' }],
+    payable: false,
+    stateMutability: 'view',
+    type: 'function'
+  }
+];
```
Comment (ChaituVR) [inline src/strategies/unipool-same-token/index.ts:29]:
```suggestion
const contractAbi = [
  'function balanceOf(address account) view returns (uint256)',
  'function earned(address account) view returns (uint256)'
];
```

File: src/strategies/unipool-same-token/examples.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        "symbol": "TOKN",
+        "decimals": 18,
+        "unipoolAddress": "0xb9840A4A8a671f79DE3Df3b812fEEb38047CE552"
+      }
+    },
+    "network": "1",
+    "addresses": [
+      "0x09212c58107c8da21b0f67a63e2144bb68bee4eb",
+      "0x8506d946CC63d1F1F3a303d68B0Da64597Cd64F3",
+      "0xc5df4d5ed23f645687a867d8f83a41836fcf8811"
```
Comment (ChaituVR) [inline src/strategies/unipool-same-token/examples.json:16]: ```✕ File examples.json should include at least 1 address with a positive score (3 ms)```
 
Please add a token address here, which can return a positive score

File: src/strategies/unipool-same-token/examples.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    ],
+    "snapshot": "11635000"
```
Comment (ChaituVR) [inline src/strategies/unipool-same-token/examples.json:18]: ```✕ File examples.json must use a snapshot block number in the past (3 ms) ```
should be a number
```suggestion
    "snapshot": 11635000
```

**Review by ChaituVR — CHANGES_REQUESTED** [review]

**Review by ChaituVR — COMMENTED** [review]
LGTM

**Comment by dapplion** [comment]
Kind bump

**Comment by dapplion** [comment]
Kind bump

**Review by bonustrack — APPROVED** [review]

**Pull request merged**